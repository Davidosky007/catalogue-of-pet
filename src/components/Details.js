import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { navigate } from '@reach/router';
import petFinder from '../helpers/petFinderApi';

const Details = ({ id }) => {
  const [loading, setLoading] = useState(true);
  const [url, setUrl] = useState('');
  const [name, setName] = useState('');
  const [animal, setAnimal] = useState('');
  const [location, setLocation] = useState('');
  const [breed, setBreed] = useState('');
  const [media, setMedia] = useState('');
  const [description, setDescription] = useState('');

  useEffect(async () => {
    const { animal } = await petFinder.animal
      .show(id)
      .then((data) => data.data);

    setUrl(animal.url);
    setName(animal.name);
    setAnimal(animal.type);
    setLocation(
      `${animal.contact.address.city},${animal.contact.address.state}`,
    );
    setBreed(animal.breeds.primary);
    setMedia(animal.photos);
    setDescription(animal.description);
    setLoading(false);
  }, [id]);

  if (loading) {
    return (
      <h1 className="loading__wrapper" data-testid="details-loading">
        Loading....
      </h1>
    );
  }

  let hero = 'http://placecorgi.com/300/300';

  if (media.length !== 0) {
    hero = media[0].large;
  }

  const home = () => navigate('/');
  const adopt = () => navigate(url);

  return (
    <div className="pets__details-wrapper">
      <div className="pets__details-wrapper-image">
        <img src={hero} alt="" />
      </div>
      <div className="pets__details-wrapper-box">
        <h1>{name}</h1>
        <h3>{`🏡 ${location}`}</h3>
        <h4>{`${animal}⭐${breed}`}</h4>
        <h2>{`Meet ${name}🎉`}</h2>
        <p>{`👀 ${description}`}</p>
        <button type="button" onClick={adopt}>
          {`Adopt ${name}`}
        </button>
        <button type="button" onClick={home}>
          Back to home
        </button>
      </div>
    </div>
  );
};

Details.propTypes = {
  id: PropTypes.number,
};

Details.defaultProps = {
  id: null,
};

export default Details;
