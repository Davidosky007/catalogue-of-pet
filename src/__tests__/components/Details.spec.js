import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Details from '../../components/Details';

test('should render text Loading', () => {
  render(<Details />);
  const details = screen.getByTestId('details-loading');
  expect(details).toHaveTextContent('Loading....');
});

test('should render Details components properly ', () => {
  const details = renderer.create(<Details />);
  const final = details.toJSON();
  expect(final).toMatchSnapshot();
});
