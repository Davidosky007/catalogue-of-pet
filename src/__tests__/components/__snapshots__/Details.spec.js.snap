// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Details components properly  1`] = `
<h1
  className="loading__wrapper"
  data-testid="details-loading"
>
  Loading....
</h1>
`;
